File: Modules/ServerTypes/Announcement/AnnouncementOccupationBindedForm.ts
import { IntegerEditor, DateEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface AnnouncementOccupationBindedForm {
    AnnouncementWizardId: IntegerEditor;
    OccupationId: IntegerEditor;
    InsertDate: DateEditor;
    UpdateDate: DateEditor;
    DeleteDate: DateEditor;
    IsActive: IntegerEditor;
    InsertUserId: IntegerEditor;
    UpdateUserId: IntegerEditor;
    DeleteUserId: IntegerEditor;
}

export class AnnouncementOccupationBindedForm extends PrefixedContext {
    static formKey = 'Announcement.AnnouncementOccupationBinded';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!AnnouncementOccupationBindedForm.init)  {
            AnnouncementOccupationBindedForm.init = true;

            var w0 = IntegerEditor;
            var w1 = DateEditor;

            initFormType(AnnouncementOccupationBindedForm, [
                'AnnouncementWizardId', w0,
                'OccupationId', w0,
                'InsertDate', w1,
                'UpdateDate', w1,
                'DeleteDate', w1,
                'IsActive', w0,
                'InsertUserId', w0,
                'UpdateUserId', w0,
                'DeleteUserId', w0
            ]);
        }
    }
}

File: Modules/HumanResource/Announcement/AnnouncementOccupationBinded/AnnouncementOccupationBindedForm.cs
using Serenity.ComponentModel;
using System;

namespace HRMSoftware.Announcement.Forms;

[FormScript("Announcement.AnnouncementOccupationBinded")]
[BasedOnRow(typeof(AnnouncementOccupationBindedRow), CheckNames = true)]
public class AnnouncementOccupationBindedForm
{
    public int AnnouncementWizardId { get; set; }
    public int OccupationId { get; set; }
    public DateTime InsertDate { get; set; }
    public DateTime UpdateDate { get; set; }
    public DateTime DeleteDate { get; set; }
    public short IsActive { get; set; }
    public int InsertUserId { get; set; }
    public int UpdateUserId { get; set; }
    public int DeleteUserId { get; set; }
}